lib/asset_sync/config.rb

```diff
@@ -224,28 +224,21 @@ def fog_options
 
     # @api
     def add_local_file_paths(&block)
-      @additional_local_file_paths_procs =
-        additional_local_file_paths_procs + [block]
+      @additional_local_file_paths_procs << block
     end
 
     # @api private
     #   This is to be called in Storage
     #   Not to be called by user
     def additional_local_file_paths
-      return [] if additional_local_file_paths_procs.empty?
-
       # Using `Array()` to ensure it works when single value is returned
-      additional_local_file_paths_procs.each_with_object([]) do |proc, paths|
+      @additional_local_file_paths_procs.each_with_object([]) do |proc, paths|
         paths.concat(Array(proc.call))
       end
     end
 
   private
 
-    # This is a proc to get additional local files paths
-    # Since this is a proc it won't be able to be configured by a YAML file
-    attr_reader :additional_local_file_paths_procs
```

Comment:
I am using an attribute reader since I want a place to document internal attribute